libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapLoader.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.maps.MapView
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+internal class MapLoader {
+
+    private val _mapViewAttached: MutableStateFlow<MapView?> = MutableStateFlow(null)
+    val mapViewAttached: StateFlow<MapView?> = _mapViewAttached.asStateFlow()
+    private val _mapViewDetached: MutableStateFlow<MapView?> = MutableStateFlow(null)
+    val mapViewDetached: StateFlow<MapView?> = _mapViewDetached.asStateFlow()
+
+    fun updateMapView(mapView: MapView) {
+        _mapViewDetached.value = mapViewAttached.value
```

Comment:
Am thinking about this 🤔 

This will mean the `onMapViewDetached` will be triggered even before `onMapViewAttached` in some cases right?

``` kotlin
navigationView.customizeMapView(nonNullMap)
navigationView.customizeMapView(null)
navigationView.addListener(listener)
```

the listener will receive an `onMapViewDetached` with the old `MapView` right? Thinking about this should be impossible (it's like receiving a Activity.onStop without an Activity.onStart)

Reply:
What do you think about 
```
private var mapViewAttached: MapView? = null
    private var mapViewDetached: MapView? = null

    private val _mapViewUpdated: MutableStateFlow<Pair<MapView?, MapView?>?> =
        MutableStateFlow(null)
    val mapViewUpdated: StateFlow<Pair<MapView?, MapView?>?> = _mapViewUpdated.asStateFlow()

    fun updateMapView(mapView: MapView) {
        mapViewDetached = mapViewAttached
        mapViewAttached = mapView
        _mapViewUpdated.value = Pair(mapViewDetached, mapViewAttached)
    }
```
This way there is no concept of `attached` or `detached`. But you always get `oldMapView, newMapView` when ever a new instance of map view is created.
@kmadsen

Reply:
> fun onMapViewUpdated(old: MapView?, new: MapView?)

Still prefer to make it consistent with [CarMapObserver](https://github.com/mapbox/mapbox-maps-android/blob/main/extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapObserver.kt) and [MapboxNavigationObserver](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationObserver.kt) so that we can build integrations that work consistently.

The `onAttached` `onDetached` is motivated by all other Android lifecycle events (e.g., onStart/onStop)

Reply:
another reason i prefer separate functions, is because you always need separate logic for them.

if it is a single function, your implementation will always be `if (old != new)` or `if (old != null)` or `if (new != null)`.